DTML.EduBot/Dialogs/RootDialog.cs

```diff
@@ -3,76 +3,265 @@
+    using DTML.EduBot.UserData;
 
     [Serializable]
     public class RootDialog : IDialog<string>
     {
+        private readonly int MAX_ATTEMPT = 2;
+
+        private const string Yes = "Yes";
+        private const string No = "No";
+
+        private static readonly IEnumerable<string> YesNoChoices = new ReadOnlyCollection<string>
+            (new List<String> {
+                Yes,
+                No});
+
         private const string ChatWithBot = "Chat With Bot";
-        private const string StartTheLessonPlan = "Start The Lesson Plan";
+        private const string StartTheLessonPlan = "Start English Lesson Plan";
 
-        private static readonly IEnumerable<string> Choices = new ReadOnlyCollection<string>
+        private static readonly IEnumerable<string> DialogChoices = new ReadOnlyCollection<string>
             (new List<String> {
                 ChatWithBot,
                 StartTheLessonPlan});
 
+        private const string TooManyAttemptMessage = "Sorry, you have attempted too many times :(";
+
         public Task StartAsync(IDialogContext context)
         {
             context.Wait(this.MessageReceivedAsync);
             return Task.CompletedTask;
         }
 
-        private Task MessageReceivedAsync(IDialogContext context, IAwaitable<IMessageActivity> result)
+        private async Task MessageReceivedAsync(IDialogContext context, IAwaitable<IMessageActivity> result)
         {
-            string friendlyUserName = context.Activity.From.Name;
+            var userText = (await result).Text;
             
+            string detectedLanguageIsoCode = await MessageTranslator.IdentifyLangAsync(userText);
+
+            using (var scope = WebApiApplication.FindContainer().BeginLifetimeScope())
+            {
+                UserData.UserData userData = new UserData.UserData();
+                userData.UserId = context.Activity.From.Id;
+                userData.NativeLanguageIsoCode = detectedLanguageIsoCode;
+
+                scope.Resolve<IUserDataRepository>().UpdateUserData(userData);
+
+                if (MessageTranslator.DEFAULT_LANGUAGE.Equals(detectedLanguageIsoCode))
+                {
+                    await context.PostAsync(BotPersonality.UserNameQuestion);
+
+                    // detected it's english language
+                    context.Wait(this.UserNameReceivedAsync);
+                }
+                else
+                {
+                    // detected user language
+
+                    CultureInfo[] allCultures =CultureInfo.GetCultures(CultureTypes.AllCultures);
+                    CultureInfo detectedCulture =
+                       allCultures
+                            .Where(cultureInfo => (detectedLanguageIsoCode.Contains(cultureInfo.TwoLetterISOLanguageName)))
+                            .FirstOrDefault();
+
+                    string detectedLanguageName = CultureInfo.GetCultureInfo(MessageTranslator.DEFAULT_LANGUAGE).DisplayName;
```

Comment:
This is getting the language name of DEFAULT_LANGUAGE, not the detected culture. Shouldn't this simply be `detectedCulture.DisplayName`